packages/api-graphql/src/GraphQLAPI.ts

```diff
@@ -38,10 +53,33 @@ export class GraphQLAPIClass extends InternalGraphQLAPIClass {
 		options: GraphQLOptions,
 		additionalHeaders?: CustomHeaders,
 	): Observable<GraphQLResult<T>> | Promise<GraphQLResult<T>> {
-		return super.graphql(amplify, options, additionalHeaders, {
-			category: Category.API,
-			action: ApiAction.GraphQl,
-		});
+		let cleanOptions = options;
+		let userAgentDetails: CustomUserAgentDetails;
+
+		if (isGraphQLOptionsWithOverride(options)) {
+			const {
+				[INTERNAL_USER_AGENT_OVERRIDE]: internalUserAgentOverride,
+				...rest
+			} = options;
+			userAgentDetails = {
+				category: Category.API,
+				action: ApiAction.GraphQl,
+				...internalUserAgentOverride,
+			};
+			cleanOptions = rest;
+		} else {
+			userAgentDetails = {
+				category: Category.API,
+				action: ApiAction.GraphQl,
+			};
+		}
+
+		return super.graphql(
+			amplify,
+			cleanOptions,
+			additionalHeaders,
+			userAgentDetails,
+		);
```

Comment:
Might be able to tighten this up and avoid usage of `let`
```suggestion
		const userAgentDetails: CustomUserAgentDetails = {
			category: Category.API,
			action: ApiAction.GraphQl,
		};

		if (isGraphQLOptionsWithOverride(options)) {
			const {
				[INTERNAL_USER_AGENT_OVERRIDE]: internalUserAgentOverride,
				...cleanOptions
			} = options;

			return super.graphql(amplify, cleanOptions, additionalHeaders, {
				...userAgentDetails,
				...internalUserAgentOverride,
			});
		}

		return super.graphql(amplify, options, additionalHeaders, {
			...userAgentDetails,
		});
	}
```